### PR TITLE
Get proxy workload label from specified cluster

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/serviceregistry/aggregate/controller.go [hunk 1]
```diff
@@ -246,12 +246,21 @@ func (c *Controller) GetProxyServiceInstances(node *model.Proxy) []*model.Servic
...
 func (c *Controller) GetProxyWorkloadLabels(proxy *model.Proxy) labels.Collection {
 	var out labels.Collection
-	// It doesn't make sense for a single proxy to be found in more than one registry.
-	// TODO: if otherwise, warning or else what to do about it.
+	clusterID := nodeClusterID(proxy)
 	for _, r := range c.GetRegistries() {
-		wlLabels := r.GetProxyWorkloadLabels(proxy)
-		if len(wlLabels) > 0 {
-			out = append(out, wlLabels...)
+		if clusterID == "" {
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/aggregate/controller.go:251]: nit: Can you combine these two single if and add a comment?

**Review by ramaraochavali — APPROVED** [review]

**Review by howardjohn — COMMENTED** [review]
This seems right but curious if this caused any issues?

**Comment by hzxuzhonghu** [comment]
In normal k8s env, as proxies are carrying labels in metadata, this function can not be run into. I think it can only happen on old ones.

**Pull request merged**